### PR TITLE
Correct offenses for `InternalAffairs/RedundantSourceRange`

### PR DESCRIPTION
**Pull request by ydah**
https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#logs

```
Inspecting 278 files
............C...........C....................C.........................C.........................C.C..................................................................................................................................................................................

Offenses:

lib/rubocop/cop/rspec/be_nil.rb:60:36: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            corrector.replace(node.source_range, 'be(nil)')
                                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/be_nil.rb:68:36: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            corrector.replace(node.source_range, 'be_nil')
                                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/contain_exactly.rb:50:36: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            corrector.replace(node.source_range, 'be_empty')
                                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/contain_exactly.rb:6[7](https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#step:5:8):1[8](https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#step:5:9): C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            node.source_range,
                 ^^^^^^^^^^^^
lib/rubocop/cop/rspec/expect_actual.rb:[9](https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#step:5:10)6:36: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
          corrector.replace(actual.source_range, expected.source)
                                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/expect_actual.rb:97:38: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
          corrector.replace(expected.source_range, actual.source)
                                     ^^^^^^^^^^^^
lib/rubocop/cop/rspec/match_array.rb:57:36: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            corrector.replace(node.source_range, 'eq([])')
                                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/match_array.rb:67:20: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
              node.source_range,
                   ^^^^^^^^^^^^
lib/rubocop/cop/rspec/predicate_matcher.rb:[10](https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#step:5:11)2:21: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
            matcher.source_range,
                    ^^^^^^^^^^^^
lib/rubocop/cop/rspec/predicate_matcher.rb:217:37: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
          corrector.replace(matcher.source_range, replacement_matcher)
                                    ^^^^^^^^^^^^
lib/rubocop/cop/rspec/predicate_matcher.rb:229:41: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
          corrector.insert_after(actual.source_range,
                                        ^^^^^^^^^^^^
lib/rubocop/cop/rspec/rails/have_http_status.rb:41:51: C: [Correctable] InternalAffairs/RedundantSourceRange: Remove the redundant source_range.
                corrector.replace(response_status.source_range, 'response')
                                                  ^^^^^^^^^^^^

278 files inspected, [12](https://github.com/rubocop/rubocop-rspec/actions/runs/4464125341/jobs/7839977882?pr=1605#step:5:13) offenses detected, 12 offenses autocorrectable
Error: Process completed with exit code 1.
```

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [-] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Review by pirj — APPROVED** [review]
Nice! Thank you. I guess this would be a relief when dealing with the LineLength cop.

Wondering, was it available since 1.0? Is it ok to use that with our minimal ‘rubocop’ version constraint?

**Comment by bquorning** [comment]
[It looks like](https://github.com/rubocop/rubocop/commit/528f3973ab0533020fb067771963ceb1e43ec94b) it was added in v1.47.0. It should be trivial to test with our lowest supported RuboCop version and see if the changes on this branch cause the specs to fail.

File: lib/rubocop/cop/rspec/predicate_matcher.rb [hunk 1]
```diff
@@ -226,7 +226,7 @@ def move_predicate(corrector, actual, matcher, block_child)
           block = block_loc ? block_loc.source : ''
 
           corrector.remove(block_loc) if block_loc
-          corrector.insert_after(actual.source_range,
+          corrector.insert_after(actual,
                                  ".#{predicate}" + args + block)
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/predicate_matcher.rb:230]: Probably this will fit on one line now?

**Review by bquorning — APPROVED** [review]
The changes seem to work fine with RuboCop v1.41.0, the lowest version that we can bundle with.

(Our gemspec says we can bundle with RuboCop `~> 1.33`, but since we have a dependency on rubocop-capybara which requires RuboCop `~> 1.41`, we inherit that same restriction.)

File: lib/rubocop/cop/rspec/predicate_matcher.rb (hunk 1 above)
Comment (bquorning) [inline lib/rubocop/cop/rspec/predicate_matcher.rb:230]:
```suggestion
          corrector.insert_after(actual, ".#{predicate}" + args + block)
```

**Pull request merged**